### PR TITLE
LINK-641, LINK-645, LINK-646, LINK-647 | Integrate enrolment related pages with LE API

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Integrate enrolment listing page with Linked Events API
Integrate create enrolment page with Linked Events API
Integrate edit enrolment page with Linked Events API
Showing warning if trying to add new enrolment to a registration where enrolment is closed

## Closes 
[LINK-641](https://helsinkisolutionoffice.atlassian.net/browse/LINK-641)
[LINK-645](https://helsinkisolutionoffice.atlassian.net/browse/LINK-645)
[LINK-646](https://helsinkisolutionoffice.atlassian.net/browse/LINK-646)
[LINK-647](https://helsinkisolutionoffice.atlassian.net/browse/LINK-647)

## Screenshots
Screenshot 2021-12-02 at 14 08 12

Screenshot 2021-12-02 at 14 12 13

Screenshot 2021-12-02 at 14 09 41

## Testing
Enrolment test page can be tested e.g. at http://localhost:3000/fi/registrations/11/enrolments
Creating enrolment can be tested e.g. at http://localhost:3000/fi/registrations/14/enrolments/create
Editing enrolment can be tested e.g. at http://localhost:3000/fi/registrations/14/enrolments/edit/3
Closed enrolment can be tested e.g. at http://localhost:3000/fi/registrations/11/enrolments/create

### CODE REVIEW

**Review by Sartonon — APPROVED** [review]
Code looking good. Some minor comments but nothing that is a must change.

Some issues:

Seems like there is no notification when editing enrolment and saving:

Screenshot 2021-12-13 at 10 59 58

Table row columns cannot be navigated with screen reader if they hava role="button"

Screenshot 2021-12-13 at 9 53 49

Also wondering what is the benefit of having the whole form visible even when enrolment is closed:

Screenshot 2021-12-13 at 13 14 25

File: src/domain/app/i18n/en.json [hunk 1]
```diff
@@ -186,31 +186,32 @@
     "pageTitle": "Edit participant"
   },
   "enrolment": {
+    "attendeeStatus": {
+      "attending": "Attendeing",
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/app/i18n/en.json]: Fixed

File: src/domain/enrolment/EditEnrolmentPage.tsx [hunk 2]
```diff
@@ -30,25 +35,49 @@ import EnrolmentFormFields from './enrolmentFormFields/EnrolmentFormFields';
...
+      onError: (error: any) => showServerErrors({ error }),
+      onSuccess: async () => {
+        await refetch();
+        window.scrollTo(0, 0);
+      },
+    });
+  };
+
   return (
     <Formik
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/enrolment/EditEnrolmentPage.tsx:82]: Fixed

File: src/domain/enrolment/EditEnrolmentPage.tsx [hunk 3]
```diff
@@ -90,6 +120,7 @@ const EditEnrolmentPage: React.FC<Props> = ({
                 >
                   <FormContainer>
                     <AuthRequiredNotification />
+                    <ServerErrorSummary errors={serverErrorItems} />
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/enrolment/EditEnrolmentPage.tsx:123]: Ahh, forgot to add translations for new server error types. Fixed

File: src/domain/enrolment/createButtonPanel/CreateButtonPanel.tsx [hunk 4]
```diff
@@ -35,7 +41,19 @@ const CreateButtonPanel: React.FC<CreateButtonPanelProps> = ({
...
+            saving ? (
+              <LoadingSpinner
+                className={styles.loadingSpinner}
+                isLoading={true}
+                small={true}
+              />
+            ) : (
+              <IconPen aria-hidden={true} />
+            )
+          }
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/enrolment/createButtonPanel/CreateButtonPanel.tsx]: Added reusable button component with icon and saving props

**Pull request merged**